src/lib/syncPagePath.ts

```diff
@@ -0,0 +1,411 @@
+import type { PageMetadata } from './readPage.js';
+import type { GuidesRequestRepresentation } from './types.js';
+import type DocsUploadCommand from '../commands/docs/upload.js';
+
+import fs from 'node:fs/promises';
+import path from 'node:path';
+
+import chalk from 'chalk';
+import ora from 'ora';
+import toposort from 'toposort';
+
+import { APIv2Error } from './apiError.js';
+import { fix, writeFixes } from './frontmatter.js';
+import isCI from './isCI.js';
+import { oraOptions } from './logger.js';
+import promptTerminal from './promptWrapper.js';
+import readdirRecursive from './readdirRecursive.js';
+import { fetchMappings, fetchSchema } from './readmeAPIFetch.js';
+import readPage from './readPage.js';
+import { categoryUriRegexPattern, parentUriRegexPattern } from './types.js';
+
+/**
+ * Commands that use this file for syncing Markdown via APIv2.
+ *
+ * Note that the `changelogs` command is not included here
+ * because it is backed by APIv1.
+ */
+export type CommandsThatSyncMarkdown = DocsUploadCommand;
+
+type PageRepresentation = GuidesRequestRepresentation;
+
+interface FailedPushResult {
+  error: APIv2Error | Error;
+  filePath: string;
+  result: 'failed';
+  slug: string;
+}
+
+type PushResult =
+  | FailedPushResult
+  | {
+      filePath: string;
+      response: PageRepresentation | null;
+      result: 'created' | 'skipped' | 'updated';
+      slug: string;
+    };
+
+/**
+ * Reads the contents of the specified Markdown or HTML file
+ * and creates/updates the corresponding page in ReadMe
+ */
+async function pushPage(
```

Comment:
Might be worth renaming this and `PageMetadata` to be about a "file" instead of a "page" so it's a little clearer as the only thing we really use the "page" nomenclature around is are custom pages.

Reply:
ah so i actually intentionally went with "page" since that's the nomenclature we use in APIv2 code for guides and reference pages. "file" felt a bit too vague since this is only concerned with markdown and not openapi